README.md

```diff
@@ -14,10 +14,57 @@ pipelines.
 
 Each `Task` is provided in a separate directory along with a README.md and a
 Kubernetes manifest, so you can choose which `Task`s to install on your
-cluster. A directory can hold more than one task (e.g. [`golang`](golang)).
+cluster. A directory can hold one task and multiple versions.
 
 _See [our project roadmap](roadmap.md)._
 
+## Catalog Structure
+
+1. Each resource follows the following structure
+
+    ```
+    ./task/                     👈 the kind of the resource
+
+        /argocd                 👈 definition file must have same name
+           /0.1
+             /OWNERS            👈 owners of this resource
+             /README.md
+             /argocd.yaml       👈 the file name should match the resource name
+             /samples/deploy-to-k8s.yaml
```

Comment:
i think some of the folders in this PR use "examples" not "samples" 🤔  i'm happy with either, as long as we're consistent :D